packages/cel/src/access.ts

```diff
@@ -621,9 +664,7 @@ export class ConcreteAttributeFactory implements AttributeFactory {
         if (val instanceof EvalAttr) {
           return new EvalAccess(id, val, this, opt);
         }
-        if (typeof val === "string") {
-          return new StringAccess(id, val, val, opt);
-        } else if (val instanceof CelUint) {
```

Comment:
Something I happen to notice, not related to `BoolAccess`. This is always false as we change type to be "object" just 3 lines above this.